tests/random_attack.py

```diff
@@ -707,6 +739,23 @@ def add_range_column(self, name, rangeobj):
         self.dedup_names()
         self.df.cbind(dt.Frame(rangeobj, names=[name]))
 
+    def set_key_columns(self, keys, names):
+        try:
+            self.df.key = names
+        except ValueError as e:
```

Comment:
Same here: you should know in advance whether setting the key is valid or not,
and then either set the key (when it's valid),
or check that a proper exception is thrown (if it's invalid).